### PR TITLE
feat(vm-route-forge): add route interface

### PR DESCRIPTION
**Pull request by yaroslavborbat**
## Description
add route Watcher interface (Watcher tracks routes on the host)
add a new Watcher implementation - ticker

## Why do we need it, and what problem does it solve?
We needs to track routes on the host. Therefore, a new interface is being introduced. Now there are 2 implementations, ticker and netlink

## Checklist
- [ ] The code is covered by unit tests.
- [ ] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [X] Changes were tested in the Kubernetes cluster manually.

### CODE REVIEW

File: images/vm-route-forge/internal/netlinkmanager/manager.go [hunk 1]
```diff
@@ -258,7 +258,7 @@ func (m *Manager) UpdateRoute(vm *virtv2.VirtualMachine) error {
 	if len(nodeIPx) == 0 {
 		return fmt.Errorf("invalid IP address %s", nodeIP)
 	}
-	m.cache.Set(vmKey, cache2.Addresses{VMIP: net.ParseIP(vmIP), NodeIP: net.ParseIP(nodeIP)})
+	m.cache.Set(vmKey, cache2.Addresses{VMIP: cache2.IP(vmIP), NodeIP: cache2.IP(nodeIP)})
```
Comment (Isteb4k) [inline images/vm-route-forge/internal/netlinkmanager/manager.go:261]: Rename cache2

File: images/vm-route-forge/internal/controller/route/watch.go [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+	"vm-route-forge/internal/cache"
+)
+
+const (
+	defaultChanSize = 100
+)
+
+type Watcher interface {
+	Watch(ctx context.Context) (chan types.NamespacedName, error)
```
Comment (Isteb4k) [inline images/vm-route-forge/internal/controller/route/watch.go:35]: (<-chan types.NamespacedName, error)

File: images/vm-route-forge/internal/controller/route/ticker.go [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+	"vm-route-forge/internal/cache"
+	"vm-route-forge/internal/netlinkwrap"
+)
+
+func NewTickerWatcher(cidrs []*net.IPNet, cache cache.Cache, tableID int, log logr.Logger) *TickerWatcher {
+	return &TickerWatcher{
+		ch:        make(chan types.NamespacedName, defaultChanSize),
+		cidrs:     cidrs,
+		cache:     cache,
+		tableID:   tableID,
```
Comment (Isteb4k) [inline images/vm-route-forge/internal/controller/route/ticker.go:38]: tableID -> routeTableID

File: images/vm-route-forge/internal/controller/route/ticker.go [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+	}
+}
+
+type TickerWatcher struct {
+	ch        chan types.NamespacedName
+	cidrs     []*net.IPNet
+	cache     cache.Cache
+	tableID   int
+	log       logr.Logger
+	routeGet  func(net.IP) ([]netlink.Route, error)
```
Comment (Isteb4k) [inline images/vm-route-forge/internal/controller/route/ticker.go:51]: Why store it in the struct's field instead of creating a routeGet() method?

**Review by Isteb4k — CHANGES_REQUESTED** [review]

**Review by Isteb4k — DISMISSED** [review]

**Review by Isteb4k — APPROVED** [review]

**Comment by fl64** [comment]
Need more information on what problem the changes to this PR are solving

**Pull request merged**